### PR TITLE
fix(es/minifier): Don't drop used variables

### LINKED ISSUE #2262: minify remove necessary code
Example:

https://github.com/SoloJiang/minify-remove-necessary-code

You can start a http-server in the lib dir, and see `esm is not defined`, but you can see the source, esm is defined.

### PR DESCRIPTION
**Pull request by fabiosantoscode**
This reproduces #2262, though I'm not sure how to fix

---

Closes #2262

### CODE REVIEW

**Comment by kdy1** [comment]
Thank you a lot!
Reducing the error case is the hardest part.
May I send some PR to your fork?

File: ecmascript/minifier/tests/compress/fixture/issues/2262/1/input.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+(() => {
+    "use strict";
+
+    var commonjsGlobal = globalThis
+
+    function createEventEmitter(value) {
+    }
```
Comment (fabiosantoscode) [inline ecmascript/minifier/tests/compress/fixture/issues/2262/1/input.js:7]: Removing this `createEventEmitter` or `commonjsGlobal` (both of which are unused) will cause the bug not to reproduce. No idea why, though

File: ecmascript/minifier/tests/compress/fixture/issues/2262/1/input.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    var index = somethingGlobal;
+
+    const esm = index;
```
Comment (fabiosantoscode) [inline ecmascript/minifier/tests/compress/fixture/issues/2262/1/input.js:11]: This indirection is important to the bug. `const esm = somethingGlobal `wouldn't reproduce it either.

**Comment by fabiosantoscode** [comment]
> Reducing the error case is the hardest part.

I have lots of practice by now :D

**Comment by fabiosantoscode** [comment]
Invited you to my fork @kdy1

**Comment by kdy1** [comment]
Thanks!

**Comment by fabiosantoscode** [comment]
Also @kdy1 is there a practical way to repeatedly run the SWC minifier against something? I managed to do that by repeatedly using the `./scripts/run.sh` with a written test, but I'm really used to the flow I had in Terser, where I'd have some file (usually bug.js) and then run Terser repeatedly on it using the CLI

**Comment by fabiosantoscode** [comment]
Maybe I can contribute a `[bin]` target for `ecmascript/minifier` which could emulate the terser CLI options

**Comment by kdy1** [comment]
Currently there's no way to do so, but a binary seems like a good idea to me.

**Pull request merged**